src/Mod/Fem/femsolver/elmer/equations/electrostatic_writer.py

```diff
@@ -125,22 +125,26 @@ def handleElectrostaticBndConditions(self):
                     # output the FreeCAD label as comment
                     if obj.Label:
                         self.write.boundary(name, "! FreeCAD Name", obj.Label)
-                    if obj.PotentialEnabled:
-                        if hasattr(obj, "Potential"):
-                            # Potential was once a float and scaled not fitting SI units
-                            if isinstance(obj.Potential, float):
-                                savePotential = obj.Potential
-                                obj.removeProperty("Potential")
-                                obj.addProperty(
-                                    "App::PropertyElectricPotential",
-                                    "Potential",
-                                    "Parameter",
-                                    "Electric Potential",
-                                )
-                                # scale to match SI units
-                                obj.Potential = savePotential * 1e6
-                            potential = float(obj.Potential.getValueAs("V"))
-                            self.write.boundary(name, "Potential", potential)
+                    if obj.Dirichlet:
+                        if obj.PotentialEnabled:
+                            if hasattr(obj, "Potential"):
```

Comment:
These can be combined into `if obj.PotentialEnabled and hasattr(obj, "Potential"):`